### PR TITLE
 Import `SystemConfigProxy` address from new location in `superchain` (registry).

### PR DESCRIPTION
**Pull request by geoknee**
**Description**

Updates the dependency on `superchain-registry` and makes necessary changes to cope with the breaking changes from https://github.com/ethereum-optimism/superchain-registry/pull/94 .

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -220,5 +220,5 @@ require (
 
 replace github.com/ethereum/go-ethereum v1.13.8 => github.com/ethereum-optimism/op-geth v1.101308.1-rc.1
 
-//replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain-registry/superchain
+replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain-registry/superchain
```
Comment (geoknee) [inline go.mod]: This temporary change needs to be undone before merging. And a newer version of the registry imported (doesn't exist yet).

File: op-chain-ops/upgrades/check.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func GetContractVersions(ctx context.Context, addresses *superchain.AddressList,
 	var versions superchain.ContractVersions
 	var err error
 
-	versions.L1CrossDomainMessenger, err = getVersion(ctx, common.HexToAddress(addresses.L1CrossDomainMessengerProxy.String()), backend)
```
Comment (geoknee) [inline op-chain-ops/upgrades/check.go]: This is a drive by change which can be pinched off into a separate PR and then proliferated a bit more (there are many instances I didn't change yet).

File: go.mod (hunk 1 above)
Comment (sebastianst) [reply go.mod]: you can just reference the commit of your superchain-registry pr in this pr to test everything together already.
```
go get github.com/ethereum-optimism/superchain-registry/superchain@177cc47729e8d169581fc00b928a3b02c7ccde98
```

File: op-chain-ops/upgrades/check.go (hunk 2 above)
Comment (geoknee) [reply op-chain-ops/upgrades/check.go]: https://github.com/ethereum-optimism/optimism/pull/9599

File: go.mod [hunk 3]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/crate-crypto/go-kzg-4844 v0.7.0
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
 	github.com/ethereum-optimism/go-ethereum-hdwallet v0.1.3
-	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240213180600-05287996b949
+	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240214172449-177cc47729e8
```
Comment (geoknee) [inline go.mod]: Putting a placeholder comment here so we update this before merging.

File: op-node/rollup/superchain.go [hunk 4]
```diff
@@ -56,8 +56,10 @@ func LoadOPStackRollupConfig(chainID uint64) (*Config, error) {
 	}
 
 	var depositContractAddress common.Address
+	var systemConfigProxyAddress common.Address
 	if addrs, ok := superchain.Addresses[chainID]; ok {
 		depositContractAddress = common.Address(addrs.OptimismPortalProxy)
+		systemConfigProxyAddress = common.Address(addrs.SystemConfigProxy)
 	} else {
 		return nil, fmt.Errorf("unable to retrieve deposit contract address")
 	}
```
Comment (sebastianst) [inline op-node/rollup/superchain.go]: I already found this confusing before. 
```suggestion
	addrs, ok := superchain.Addresses[chainID]
	if !ok {
		return nil, fmt.Errorf("unable to retrieve deposit contract address")
	}
	depositContractAddress := common.Address(addrs.OptimismPortalProxy)
	systemConfigProxyAddress := common.Address(addrs.SystemConfigProxy)
```
Or maybe not even set the address variables and directly use `addrs`, seem to only be used once each.

**Review by sebastianst — COMMENTED** [review]
looks like a pretty small change after all, happy to do with Option A in https://github.com/ethereum-optimism/superchain-registry/pull/94

File: op-node/rollup/superchain.go (hunk 4 above)
Comment (geoknee) [reply op-node/rollup/superchain.go]: Yep this is better. Will fix.

Comment (geoknee) [reply op-node/rollup/superchain.go]: [589b602](https://github.com/ethereum-optimism/optimism/pull/9585/commits/589b602cdfc56ebb6f22f64170d8b73d69c465e1)

File: go.mod [hunk 5]
```diff
@@ -220,5 +220,6 @@ require (
 
 replace github.com/ethereum/go-ethereum v1.13.8 => github.com/ethereum-optimism/op-geth v1.101308.1-rc.1
 
-//replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain-registry/superchain
+// replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain-registry/superchain
+
```
[the remark this answers is not in the document]
Comment (geoknee) [reply go.mod:224]: I think it is a remnant of local development, but not from this PR :-P . Perhaps the reviewer can comment on whether we can remove it.

**Review by sebastianst — APPROVED** [review]
LGTM. Will merge in tandem with https://github.com/ethereum-optimism/superchain-registry/pull/94

**Review by protolambda — CHANGES_REQUESTED** [review]
Why isn't the `mainnetCfg` sanity test in the op-node chain config package failing? I think the ecotone time in there should reflect the change.

**Comment by sebastianst** [comment]
> Why isn't the `mainnetCfg` sanity test in the op-node chain config package failing? I think the ecotone time in there should reflect the change.

It's being upgraded here https://github.com/ethereum-optimism/optimism/pull/9625 this one pulls in an earlier version of the superchain-registry.

**Review by protolambda — APPROVED** [review]
Correction: nevermind the ecotone time test. This updates the registry to just before that change happened, so no need to update the test yet.

**Review by trianglesphere — APPROVED** [review]
